### PR TITLE
MM-8681 Instructions for setting up Mattermost to run on CloudFront

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

File: source/install/config-cloudfront.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.. _config-cloudfront:
+
+Configuring CloudFront to host Mattermost static assets
+=======================================================
+
+1. Create an S3 bucket using your desired domain. In our example it will be mattermost.example.com.
+2. Enable static hosting for you S3 bucket.
+3. From the Mattermost distribution, upload the ``client`` directory to s3. Renaming it to ``static``. You can use aws CLI command below from within the client directory. Be sure however you do this that the files are public readable.
```
Comment (jasonblais) [inline source/install/config-cloudfront.rst]: what's the purpose of? `You can use aws CLI command below from within the client directory. Be sure however you do this that the files are public readable.`

File: source/install/config-cloudfront.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    ``aws s3 cp --acl public-read --recursive . s3://static.spinmint.com/static/``
+
+4. Setup your Mattermost app server and create a record from a subdomain of your desired domain to point directly to your app server. This is to bypass CloudFront to connect websockets. For our example we will use the domain ws.mattermost.example.com. If you have multiple app servers, this domain should point to the load balancer/proxy such as ALB or nginx.
```
Comment (jasonblais) [inline source/install/config-cloudfront.rst]: ALB or ELB?

File: source/install/config-cloudfront.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+   c. Set allowed HTTP Methods to GET, HEAD, OPTIONS, PUT, POST, PATCH, DELETE.
+   d. Forward Cookies should be set to All.
+   e. Query string forwarding and Caching should be set to Forward all, cache based on all.
+8. Next setup some custom error responses under the Error Pages tab.
+   a. Set HTTP Error Code to 403.
+   b. Set Customize Error Response to Yes.
+   c. Set response page path to ``/static/root.html``.
+   d. Set HTTP Response Code to 200.
+   e. Create and do the same for the HTTP Error Code 404.
+9. Now you can setup the domain you want Mattermost to be served from to point to your CoudFront distribution.
```
Comment (jasonblais) [inline source/install/config-cloudfront.rst]: How?

File: source/install/config-cloudfront.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
+.. _config-cloudfront:
+
+Configuring CloudFront to host Mattermost static assets
+=======================================================
+
+1. Create an S3 bucket using your desired domain. In our example it will be mattermost.example.com.
+2. Enable static hosting for your S3 bucket.
+3. From the Mattermost distribution, upload the ``client`` directory to S3. Rename it to ``static``. You can use the AWS CLI command below from within the ``client`` directory. Note that the files are publicly readable, so be careful how you do it.
+
+    ``aws s3 cp --acl public-read --recursive . s3://static.spinmint.com/static/``
```
Comment (lindy65) [inline source/install/config-cloudfront.rst]: Should there be a space before and after the "period" between "--recursive" and "s3"?

File: source/install/config-cloudfront.rst (hunk 1 above)
Comment (crspeller) [reply source/install/config-cloudfront.rst]: It's just an example of how to upload all the files to s3. And the second sentence is you need to make sure that the files have the permission public-read (the `--acl public-read` part of the command)

File: source/install/config-cloudfront.rst (hunk 2 above)
Comment (crspeller) [reply source/install/config-cloudfront.rst]: ALB actually. Amazon Application Load Balancer. The newer version of ELB.

File: source/install/config-cloudfront.rst (hunk 3 above)
Comment (crspeller) [reply source/install/config-cloudfront.rst]: Depends on how you register your domains. Doing that is outside the scope of this document.

File: source/install/config-cloudfront.rst (hunk 4 above)
Comment (crspeller) [reply source/install/config-cloudfront.rst]: Yup. `.` means current directory :P

**Comment by crspeller** [comment]
@jasonblais Ready for re-review. Added motivation section and also one upgrade note.

**Pull request merged**